### PR TITLE
feat(provider/azure): Add form validation on submit

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
Rather than disabling the submit button and forcing the user to guess which fields are required, always enable the button and display errors/focus to the first error on submit

We also considered adding '*' or '(optional)' or '(required)' to field names, but found this to be the easiest to consistently apply across forms

### CODE REVIEW

**Comment by EricJizbaMSFT** [comment]
@rguthriemsft @dcaro @clguimanmsft PTAL

Note: This requires https://github.com/spinnaker/deck/pull/3545 to work properly for the account/region fields

**Comment by EricJizbaMSFT** [comment]
ping @rguthriemsft @dcaro @clguimanmsft

File: app/scripts/modules/azure/serverGroup/configure/serverGroupAdvancedSettingsSelector.directive.html [hunk 1]
```diff
@@ -49,13 +49,13 @@
   </div>
 </div>
 <div class="form-group">
-  <div class="col-md-5 sm-label-right"><b>Ramdisk Id (optional)</b></div>
```
Comment (rguthriemsft) [inline app/scripts/modules/azure/serverGroup/configure/serverGroupAdvancedSettingsSelector.directive.html]: Is this a bug that these fields are even here?  These look like AWS fields that are result of bad/copy paste.

File: app/scripts/modules/azure/serverGroup/configure/serverGroupAdvancedSettingsSelector.directive.html [hunk 2]
```diff
@@ -49,13 +49,13 @@
...
+  <div class="col-md-5 sm-label-right"><b>Ramdisk Id</b></div>
   <div class="col-md-6"><input type="text"
                                class="form-control input-sm"
                                ng-model="command.ramdiskId"/></div>
 </div>
 <div class="form-group">
-  <div class="col-md-5 sm-label-right"><b>IAM Instance Profile (optional)</b></div>
```
Comment (rguthriemsft) [inline app/scripts/modules/azure/serverGroup/configure/serverGroupAdvancedSettingsSelector.directive.html]: same as above

File: app/scripts/modules/azure/serverGroup/configure/serverGroupAdvancedSettingsSelector.directive.html (hunk 1 above)
Comment (EricJizbaMSFT) [inline app/scripts/modules/azure/serverGroup/configure/serverGroupAdvancedSettingsSelector.directive.html]: Yes very good point - that explains why I couldn't find this UI :) Removed these files. We actually use the advancedSettings pages under 'wizard/advancedSettings/'

**Comment by rguthriemsft** [comment]
LGTM

**Comment by anotherchrisberry** [comment]
just needs a rebase and we'll get it merged

**Comment by EricJizbaMSFT** [comment]
@anotherchrisberry - rebase done

**Comment by anotherchrisberry** [comment]
Sorry, I missed this - one more rebase and we'll get this merged.

**Comment by EricJizbaMSFT** [comment]
@anotherchrisberry ready again

**Pull request merged**